packages/ui/src/components/editor/properties/scene/settings/index.tsx

```diff
@@ -0,0 +1,233 @@
+    ])
+
+    if (!envmap || !loadingScreen) return null!
+
+    const editorState = getState(EditorState)
+    const sceneName = editorState.sceneName!.split('.').slice(0, -1).join('.')
+    const projectName = editorState.projectName!
+    const envmapFilename = `${sceneName}.envmap.ktx2`
+    const loadingScreenFilename = `${sceneName}.loadingscreen.ktx2`
+
+    const currentSceneDirectory = getState(EditorState).scenePath!.split('/').slice(0, -1).join('/')
+    const promises = uploadProjectFiles(
+      projectName,
+      [new File([envmap], envmapFilename), new File([loadingScreen], loadingScreenFilename)],
+      [currentSceneDirectory, currentSceneDirectory]
+    )
+
+    const [[envmapURL], [loadingScreenURL]] = await Promise.all(promises.promises)
+
+    commitProperty(SceneSettingsComponent, 'loadingScreenURL')(loadingScreenURL)
+    state.merge({
+      loadingScreenURL: null,
+      loadingScreenImageData: null,
+      uploadingLoadingScreen: false
+    })
+  }
+
+  const generateColors = () => {
+    const url = state.thumbnailURL.value ?? sceneSettingsComponent.thumbnailURL.value
+    if (!url) return
+    const image = new Image()
+    image.crossOrigin = 'Anonymous'
+    image.onload = () => {
+      const palette = getImagePalette(image)
+      if (palette) {
+        commitProperties(SceneSettingsComponent, {
+          primaryColor: palette.color,
+          backgroundColor: palette.backgroundColor,
+          alternativeColor: palette.alternativeColor
+        })
+      }
+    }
+    image.src = url
+  }
+
+  return (
+    <PropertyGroup
+      name={t('editor:properties.sceneSettings.name')}
+      description={t('editor:properties.sceneSettings.description')}
+    >
+      <InputGroup
+        name="Thumbnail"
+        label={t('editor:properties.sceneSettings.lbl-thumbnail')}
+        info={t('editor:properties.sceneSettings.info-thumbnail')}
+      >
+        <div>
+          <ImagePreviewInput value={state.thumbnailURL.value ?? sceneSettingsComponent.thumbnailURL.value} />
+
+          <Button onClick={createThumbnail}>{t('editor:properties.sceneSettings.generate')}</Button>
+          {state.uploadingThumbnail.value ? (
+            <LoadingCircle />
+          ) : (
+            <Button onClick={uploadThumbnail} disabled={!state.thumbnail.value}>
+              {t('editor:properties.sceneSettings.save')}
+            </Button>
+          )}
+        </div>
+      </InputGroup>
+      <InputGroup
+        name="Loading Screen"
+        label={t('editor:properties.sceneSettings.lbl-loading')}
+        info={t('editor:properties.sceneSettings.info-loading')}
+      >
+        <div>
+          <ImagePreviewInput value={state.loadingScreenURL.value ?? sceneSettingsComponent.loadingScreenURL.value} />
+          <Button onClick={createLoadingScreen}>{t('editor:properties.sceneSettings.generate')}</Button>
+          {state.uploadingLoadingScreen.value ? (
+            <LoadingCircle />
```

Comment:
LoadingCircle is material-ui. For a loading spinner we can do `<LoadingView spinnerOnly />`